### PR TITLE
[Refactor] define type length constants in thrift

### LINKED ISSUE #6482: Refactor the chaotic code about type length
## Refactoring request

**Is your refactoring request related to a problem? Please describe.**

Several requirements for type length:
- What's the default byte size of a type?
- What's the estimated byte size of a column, operator? 
- What's the relative-precise row length of an operator? 

However, current code is very chaotic:
- Default type size is defined at multiple places, and not consistent
- Default type size is a static value, could not be adaptive to the real size
- The estimated type size/row length is not so precise, which lead to more secondary problems

**Describe the solution you'd like**

Several steps need to be conducted to optimize this problem:
- [ ] Remove the redundant code for default type length, and try best to make it consistent
- [ ] Leverage the statistic of column size to estimate row-length/operator-width, and expose it to BE runtime
- [ ] Make the statistic more precise, which should be adaptive to the data and its distribution

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6482

## Problem Summary(Required) ：

Remove the redundant code for default type length, and try best to make it consistent.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|

File: gensrc/thrift/Types.thrift [hunk 1]
```diff
@@ -447,3 +447,11 @@ enum TCompressionType {
...
     LZO = 11; // Deprecated
 }
+
+const i64 MAX_VARCHAR_LENGTH = 1048576;
+const i64 MAX_CHAR_LENGTH = 255;
+const i64 MAX_CHAR_INLINE_LENGTH = 128;
+const i64 DEFAULT_STRING_LENGTH = 65533;
+const i64 DEFAULT_HLL_LENGTH = 16385;
+const i64 DEFAULT_BITMAP_LENGTH = 128;
+const i64 DEFAULT_JSON_LENGTH = 128;
```
Comment (imay) [inline gensrc/thrift/Types.thrift:457]: It is very confusing that BITMAP_LENGTH/JSON_LENGTH is 128. I think this PR is your first step to refactor type length.
Later on you will give memory-size and storage-size to each kind of type?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|

File: gensrc/thrift/Types.thrift (hunk 1 above)
Comment (mofeiatwork) [reply gensrc/thrift/Types.thrift:457]: Current number `128` is from existing code. The `128` means a default size for memory estimation.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|

**Comment by mofeiatwork** [comment]
The constant generated by thrift is very hard to use, cause it's a plain global variable and has undetermined initialization order.

So temporarily close this PR, until we find a better way to fix it.

**Pull request closed without merging**